src/resources/packages/wizard/components/tabs.js

```diff
@@ -10,112 +10,216 @@ import * as TicketsContent from './tabs/tickets';
+			baseClass: 'tec-events-onboarding__tab-intro',
 			disabled: false,
-			content: IntroContent,
-			current: true
+			content: IntroContent
 		},
 		{
 			name: 'display',
 			title: __('Calendar Display', 'the-events-calendar'),
 			className: 'tec-events-onboarding__tab-display',
+			baseClass: 'tec-events-onboarding__tab-display',
 			disabled: true,
-			content: DisplayContent,
-			current: false
+			content: DisplayContent
 		},
 		{
 			name: 'settings',
 			title: __('Event Settings', 'the-events-calendar'),
 			className: 'tec-events-onboarding__tab-settings',
+			baseClass: 'tec-events-onboarding__tab-settings',
 			disabled: true,
-			content: SettingsContent,
-			current: false
+			content: SettingsContent
 		},
 		{
 			name: 'organizer',
 			title: __('Event Organizer', 'the-events-calendar'),
 			className: 'tec-events-onboarding__tab-organizer',
+			baseClass: 'tec-events-onboarding__tab-organizer',
 			disabled: true,
-			content: OrganizerContent,
-			current: false
+			content: OrganizerContent
 		},
 		{
 			name: 'venue',
 			title: __('Event Venue', 'the-events-calendar'),
 			className: 'tec-events-onboarding__tab-venue',
+			baseClass: 'tec-events-onboarding__tab-venue',
 			disabled: true,
-			content: VenueContent,
-			current: false
+			content: VenueContent
 		},
 		{
 			name: 'tickets',
 			title: __('Tickets', 'the-events-calendar'),
 			className: 'tec-events-onboarding__tab-tickets',
+			baseClass: 'tec-events-onboarding__tab-tickets',
 			disabled: true,
-			content: TicketsContent,
-			current: false
+			content: TicketsContent
 		},
 	];
 
+	/**
+	 * State management for tabs.
+	 *
+	 * @property {Array} tabs - The current state of tabs.
+	 */
 	const [tabs, setTabs] = useState(initialTabs);
+
+	/**
+	 * State management for active tab.
+	 *
+	 * @property {string} activeTab - The currently active tab.
+	 * @default 'intro'
+	 */
 	const [activeTab, setActiveTab] = useState('intro');
 
+	/**
+	 * Class names for completed tab state.
+	 *
+	 * @type {string}
+	 */
+	const completedClass = "tec-events-onboarding__tab--completed";
+
+	/**
+	 * Class names for active tab state.
+	 *
+	 * @type {string}
+	 */
+	const activeClass = "tec-events-onboarding__tab--active";
+
+	/**
+	 * "Clean" the tab classes by removing the "active" class from all tabs.
+	 */
+	const cleanTabClasses = () => {
+		const cleanedTabs = tabs.map(tab => (
+			// We just want to remove the "active" class from all tabs prior to setting the next one.
+			tab.className = tab.className.replace(activeClass, "")
+		));
+
+		return setTabs(cleanedTabs);
+	};
+
+	/**
+	 * Mark a tab as active by name.
+	 *
+	 * @param {string} tabName - The name of the tab to mark as active.
+	 */
+	const setAsActive = (tabName) => {
+		cleanTabClasses();
+
+		const activeTabs = tabs.map(tab => {
+			if (tab.name === tabName) {
+				// Add the active class to the active tab.
+				tab.className = tab.className + " " + activeClass;
+			}
+
+			return tab;
```

Comment:
Same here, need to use `map` to return a new object instance, instead of modifying the tab directly.